cmd/admin-handlers.go

```diff
@@ -1768,3 +1769,47 @@ func (a adminAPIHandlers) KMSKeyStatusHandler(w http.ResponseWriter, r *http.Req
 	}
 	writeSuccessResponseJSON(w, resp)
 }
+
+// ServerCPUHardwareInfo holds informantion about cpu hardware
+type ServerCPUHardwareInfo struct {
+	Addr  string             `json:"addr"`
+	Error string             `json:"error,omitempty"`
+	CPUHW []cpuinfo.InfoStat `json:"cpu-hw"`
+}
+
+// ServerHardwareInfoHandler - GET /minio/admin/v1/hardwareinfo?Type={hwType}
+// ----------
+// Get all hardware information based on input type
+// Supported types = cpu
+func (a adminAPIHandlers) ServerHardwareInfoHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "HardwareInfo")
+
+	objectAPI := validateAdminReq(ctx, w, r)
+	if objectAPI == nil {
+		return
+	}
+
+	vars := mux.Vars(r)
+	switch hwType := vars["hwType"]; hwType {
```

Comment:
`hwType` should be defined as a constant..

Reply:
done.